server/entitymanager/entitymanager_test.go

```diff
@@ -0,0 +1,291 @@
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			got, err := em.FetchOwnershipVoucher(test.serial)
+			if (err != nil) != test.wantErr {
+				t.Fatalf("FetchOwnershipVoucher(%v) err = %v, want %v", test.serial, err, test.wantErr)
+			}
+			if !cmp.Equal(got, test.want) {
+				t.Errorf("FetchOwnershipVoucher(%v) got %v, want %v", test.serial, got, test.want)
+			}
+		})
+	}
+}
+
+func TestResolveChassis(t *testing.T) {
+	tests := []struct {
+		desc    string
+		input   *service.EntityLookup
+		want    *service.ChassisEntity
+		wantErr bool
+	}{
+		{
+			desc: "Default device",
+			input: &service.EntityLookup{
+				SerialNumber: "123",
+				Manufacturer: "Cisco",
+			},
+			want: &service.ChassisEntity{
+				BootMode: bootz.BootMode_BOOT_MODE_SECURE,
+			},
+		},
+		{
+			desc: "Chassis Not Found",
+			input: &service.EntityLookup{
+				SerialNumber: "456",
+				Manufacturer: "Cisco",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+
+	em := New(nil).AddChassis(bootz.BootMode_BOOT_MODE_SECURE, "Cisco", "123")
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			got, err := em.ResolveChassis(test.input)
+			if (err != nil) != test.wantErr {
+				t.Fatalf("ResolveChassis(%v) err = %v, want %v", test.input, err, test.wantErr)
+			}
+			if !cmp.Equal(got, test.want) {
+				t.Errorf("ResolveChassis(%v) got %v, want %v", test.input, got, test.want)
+			}
+		})
+	}
+}
+
+func TestSign(t *testing.T) {
+	tests := []struct {
+		desc    string
+		serial  string
+		resp    *bootz.GetBootstrapDataResponse
+		wantOV  string
+		wantOC  string
+		wantErr bool
+	}{
+		{
+			desc:   "Success",
+			serial: "123A",
+			resp: &bootz.GetBootstrapDataResponse{
+				SignedResponse: &bootz.BootstrapDataSigned{
+					Responses: []*bootz.BootstrapDataResponse{
+						{SerialNum: "123A"},
+					},
+				},
+			},
+			wantOV:  "test_ov",
+			wantOC:  "test_oc",
+			wantErr: false,
+		},
+		{
+			desc:    "Empty response",
+			resp:    &bootz.GetBootstrapDataResponse{},
+			wantErr: true,
+		},
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		t.Fatal(err)
+	}
+	privPEM := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(priv),
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			artifacts := &service.SecurityArtifacts{
+				OV: service.OVList{test.serial: test.wantOV},
+				OC: &service.KeyPair{
+					Cert: test.wantOC,
+					Key:  string(pem.EncodeToMemory(privPEM)),
+				},
+			}
+			em := New(artifacts)
+
+			err := em.Sign(test.resp, test.serial)
+			if err != nil {
+				if test.wantErr {
+					t.Skip()
+				}
+				t.Errorf("Sign() err = %v, want %v", err, test.wantErr)
+			}
+			signedResponseBytes, err := proto.Marshal(test.resp.GetSignedResponse())
+			if err != nil {
+				t.Fatal(err)
+			}
+			hashed := sha256.Sum256(signedResponseBytes)
+			sigDecoded, err := base64.StdEncoding.DecodeString(test.resp.GetResponseSignature())
```

Comment:
nit: it feels to me that sign and verifySign should be in a package that could be used by both server and client code and unit tests